src/render.js

```diff
@@ -70,132 +116,63 @@ const runInContext = function(context, fn) {
-            const renderElement = (element, data) => {
-                const result = global.StyleSheetServer.renderStatic(() =>
-                    ReactDOMServer.renderToString(element));
-
-                if (data) {
-                    result.data = data;
+    const renderProfile = profile.start("rendering " + entryPointUrl);
+
+    try {
+        // Make sure the require globals are made available to the VM context.
+        if (globals) {
+            Object.keys(globals).forEach(key => {
+                // Location is a special case.
+                if (key === 'location') {
+                    context.location.replace(globals[key]);
+                } else {
+                    context[key] = globals[key];
                 }
-
-                resolve(result);
-            };
-
-            // For React elements that have no Apollo-centric logic, we just
-            // render them as normal.
-            if (!global.ApolloNetworkLink) {
-                return renderElement(reactElement);
-            }
-
-            // If network details were provided for Apollo then we go about
-            // wrapping the element in an Apollo provider (which will collect
-            // the data requirements of the child components and send off a
-            // network request to a GraphQL endpoint).
-
-            // Build an Apollo client. This is responsible for making the
-            // network requests to the GraphQL endpoint and bringing back
-            // the data.
-            const client = new global.ApolloClient.ApolloClient({
-                ssrMode: true,
-                link: global.ApolloNetworkLink,
-                cache: global.ApolloCache,
             });
+        }
 
-            // We wrap the React element with an Apollo Provider (which
-            // takes a React element and a client)
-            const wrappedElement = React.createElement(
-                global.ReactApollo.ApolloProvider,
-                {
-                    children: reactElement,
-                    client: client,
-                }
-            );
-
-            // From the Apollo Provider and the wrapped React element we
-            // use React Apollo to traverse through the full React element
-            // tree to find all the GraphQL data requirements that've been
-            // specified by the components. These requirements are pulled
-            // together and sent to the serve as a single request.
-            global.ReactApollo.getDataFromTree(wrappedElement)
-                .then(() => {
-                    // All of the data comes back as a single object which
-                    // we can then pass back to the client to be rendered
-                    // directly into the page (for the re-hydration).
-                    const initialState = client.extract();
-
-                    renderElement(wrappedElement, initialState);
-                }).catch((err) => reject(err));
-        });
-    }).then((data) => {
```

Comment:
All of this red code is now either done in the client-side entrypoint stuff and not the react-render-server, or it is done in the `performRender` method.

Reply:
This is fantastic!

One thing I realize now: it seems like this server may not even require that we use React anymore! as long as we're able to do the rendering async in a promise then we should be good!

Reply:
Exactly.